packages/go_router/example/lib/redirection.dart

```diff
@@ -74,22 +75,22 @@ class App extends StatelessWidget {
     ],
 
     // redirect to the login page if the user is not logged in
-    redirect: (GoRouterState state) {
+    redirect: (BuildContext context, GoRouterState state) {
       // if the user is not logged in, they need to login
       final bool loggedIn = _loginInfo.loggedIn;
       final bool loggingIn = state.subloc == '/login';
       if (!loggedIn) {
-        return loggingIn ? null : '/login';
+        return SynchronousFuture<String?>(loggingIn ? null : '/login');
```

Comment:
Please add a comment here explaining why `SynchronousFuture` is used instead of `Future.value`